extensions/interactions/NumberWithUnits/directives/oppia-interactive-number-with-units.component.spec.ts

```diff
@@ -16,143 +16,134 @@
-    $provide.value('$attrs', {
-      labelForFocusTarget: 'label'
-    });
-  }));
-  beforeEach(angular.mock.inject(function($injector, $componentController) {
-    $rootScope = $injector.get('$rootScope');
-    $scope = $rootScope.$new();
-    $q = $injector.get('$q');
-    $uibModal = $injector.get('$uibModal');
-    numberWithUnitsObjectFactory =
-      $injector.get('NumberWithUnitsObjectFactory');
-    currentInteractionService = $injector.get('CurrentInteractionService');
-    ctrl = $componentController('oppiaInteractiveNumberWithUnits');
-    ctrl.NumberWithUnitsForm = {
-      answer: {
-        $invalid: false,
-        $setValidity: function(errorType, valid) {
-          this.$invalid = !valid;
-        }
-      }
-    };
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [InteractiveNumberWithUnitsComponent],
+      imports: [
+        TranslateModule.forRoot({
+          useDefaultLang: true,
+          isolate: false,
+          extend: false,
+          defaultLanguage: 'en'
+        })
+      ],
+      providers: [
+        {
+          provide: CurrentInteractionService,
+          useValue: mockCurrentInteractionService
+        },
+        NumberWithUnitsObjectFactory,
+        NgbModal,
+      ],
+      schemas: [NO_ERRORS_SCHEMA]
+    }).compileComponents();
   }));
 
+  beforeEach(() => {
+    currentInteractionService = TestBed.inject(CurrentInteractionService);
+    numberWithUnitsObjectFactory = TestBed.inject(NumberWithUnitsObjectFactory);
+    ngbModal = TestBed.inject(NgbModal);
+    fixture = TestBed.createComponent(InteractiveNumberWithUnitsComponent);
+    component = fixture.componentInstance;
+  });
+
   it('should initialise component when user adds or plays interaction', () => {
     spyOn(numberWithUnitsObjectFactory, 'createCurrencyUnits');
     spyOn(currentInteractionService, 'registerCurrentInteraction');
 
-    ctrl.$onInit();
+    component.ngOnInit();
 
-    expect(ctrl.answer).toBe('');
-    expect(ctrl.labelForFocusTarget).toBe('label');
-    expect(ctrl.NUMBER_WITH_UNITS_FORM_SCHEMA).toEqual({
-      type: 'unicode',
-      ui_config: {}
-    });
+    expect(component.answer).toBe('');
     expect(numberWithUnitsObjectFactory.createCurrencyUnits).toHaveBeenCalled();
     expect(currentInteractionService.registerCurrentInteraction)
       .toHaveBeenCalled();
   });
 
-  it('should not display warning when the answer format is correct', () => {
-    ctrl.$onInit();
+  it('should not display warning when the answer format is correct',
+    fakeAsync(() => {
+      component.ngOnInit();
 
-    expect(ctrl.getWarningText()).toBe('');
+      expect(component.errorMessage).toBe('');
 
-    ctrl.answer = '24 km';
-    $scope.$apply();
+      component.answer = '24 km';
+      component.answerValueChanged();
+      tick(150);
 
-    expect(ctrl.getWarningText()).toBe('');
-  });
+      expect(component.errorMessage).toBe('');
+      expect(component.isValid).toBeTrue();
```

Comment:
Add a pre-check

Reply:
Done

Reply:
Could you set `component.isValid` to false and the check if it changes to true?